ocaml/typing/solver.ml

```diff
@@ -620,45 +603,46 @@ module Solver_mono (C : Lattices_mono) = struct
           | Amode b -> loop (C.meet obj a b) mvs xs
           (* some minor optimization: if [a] is higher than [mupper mv], we
               should keep the latter instead. This helps to fail early in
-             [submode_try] *)
+             [submode_log] *)
           | Amodevar mv ->
             loop (C.meet obj a (mupper obj mv)) (cons_dedup mv mvs) xs
           | Amodemeet (b, mvs') ->
             loop (C.meet obj a b) (rev_append_dedup mvs' mvs) xs)
     in
     loop (C.max obj) [] l
 
-  let zap_to_floor_morphvar ~commit obj mv =
-    let log, lower = zap_to_floor_morphvar_aux obj mv in
-    if commit then !append_changes log else undo_changes !log;
+  let zap_to_floor_morphvar obj mv ~log =
+    let log_, lower = zap_to_floor_morphvar_aux obj mv in
+    (match log with
+    | None -> undo_changes log_
```

Comment:
This surprises me. I would expect the `log` argument to be `None` when I don't care about logging.... but this treatment means that `None` means "be pure". I would make this more explicit. Maybe I just mean that I like the old `commit` approach.

Reply:
good point - this replaces the `commit` argument which has clearer meaning. I will rename the argument with comment.